### PR TITLE
Add operation name attribute on gateway.plan spans

### PR DESCRIPTION
**Pull request by theJC**
Addresses https://github.com/apollographql/federation/issues/2802

Add operation name to gateway.plan span

### CODE REVIEW

**Review by trevor-scheer — CHANGES_REQUESTED** [review]

File: gateway-js/src/__tests__/gateway/__snapshots__/opentelemetry.test.ts.snap [hunk 1]
```diff
@@ -83,7 +87,11 @@ exports[`opentelemetry with local data receives spans on plan failure 1`] = `
       },
       {
         "name": "gateway.plan",
-        "attributes": {},
+        "attributes": {
+          "operationName": "GetProduct",
+          "graphql.operation.name": "GetProduct",
+          "graphql.document": "#graphql\\n      subscription GetProduct($upc: String!) {\\n        product(upc: $upc) {\\n          name\\n        }\\n      }\\n    "
```
[the remark this answers is not in the document]
Comment (theJC) [reply gateway-js/src/__tests__/gateway/__snapshots__/opentelemetry.test.ts.snap:93]: great idea/feedback, thank you.

[the remark this answers is not in the document]
Comment (theJC) [reply gateway-js/src/__tests__/gateway/__snapshots__/opentelemetry.test.ts.snap]: Should now be addressed with most recent commit

**Comment by theJC** [comment]
When making changes like this in the gateway, is the expectation we also attempt to make a similar change (as close as possible) in router?

File: gateway-js/src/__tests__/gateway/__snapshots__/opentelemetry.test.ts.snap (hunk 1 above)
[the remark this answers is not in the document]
Comment (theJC) [reply gateway-js/src/__tests__/gateway/__snapshots__/opentelemetry.test.ts.snap]: @trevor-scheer  -- Anything else you'd like adjusted in this MR?  -Thanks

[the remark this answers is not in the document]
Comment (trevor-scheer) [reply gateway-js/src/__tests__/gateway/__snapshots__/opentelemetry.test.ts.snap]: Hey @theJC, should be good to go. I'll take a final pass and merge next week when I'm back from Summit.

**Review by trevor-scheer — APPROVED** [review]
Can you please run `npx changeset` and add a `minor` changeset? It can be a fairly short description explaining what's added to the span.

Thanks!

**Comment by trevor-scheer** [comment]
@theJC generally speaking, yes. I should have a better answer for you next week in this particular case.

**Comment by theJC** [comment]
Changeset added. prettier code format issue with my updates to the tests fixed as well.

**Review by trevor-scheer — APPROVED** [review]
Thanks for your work on this @theJC!

**Comment by trevor-scheer** [comment]
@theJC in this particular case, we don't intend to maintain parity with the router. The router team is currently working on a more robust solution to configuring otel.
https://github.com/apollographql/router/pull/4061

**Pull request merged**